sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,218 @@
+
+        This should only be used in special situations. See :ref:`user guide`
+        for more info.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        **fit_params : parameters to be passed to the base estimator.
+
+        Returns
+        -------
+        self : object
+
+        """
+        self.base_estimator.fit(X, y, **fit_params)
+        return self
+
+    def _method_name(self):
+        if self.method == 'auto':
+            if getattr(self.base_estimator, 'predict_proba', None):
+                method = 'predict_proba'
+            elif getattr(self.base_estimator, 'decision_function', None):
+                method = 'decision_function'
+            else:
+                method = 'predict'
+        else:
+            method = self.method
+
+        return method
+
+    def transform(self, *args, **kwargs):
+        """Transform dataset.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Input data to be transformed. Use ``dtype=np.float32`` for maximum
+            efficiency. Sparse matrices are also supported, use sparse
+            ``csr_matrix`` for maximum efficiency.
+
+        Returns
+        -------
+        X_transformed : sparse matrix, shape=(n_samples, n_out)
+            Transformed dataset.
+        """
+        t = getattr(self.base_estimator, self._method_name())
+        preds = t(*args, **kwargs)
+
+        if preds.ndim == 1:
+            preds = preds.reshape(-1, 1)
+
+        return preds
+
+    def fit_transform(self, X, y, **fit_params):
+        """Fit estimator and transform dataset.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Input data used to build forests. Use ``dtype=np.float32`` for
+            maximum efficiency.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        **fit_params : parameters to be passed to the base estimator.
+
+        Returns
+        -------
+        X_transformed : sparse matrix, shape=(n_samples, n_out)
+            Transformed dataset.
+        """
+        preds = cross_val_predict(self.base_estimator, X, y, cv=self.cv,
+                                  method=self._method_name(),
+                                  n_jobs=self.n_jobs, fit_params=fit_params)
+
+        self.base_estimator.fit(X, y, **fit_params)
+
+        if preds.ndim == 1:
+            preds = preds.reshape(-1, 1)
+
+        return preds
+
+
+def _identity(x):
+    return x
+
+
+def _identity_transformer():
+    """Contructs a transformer that returns its input unchanged"""
+    return FunctionTransformer(_identity)
```

Comment:
Maybe add `accept_sparse=True` here, or warn that `restack=True` in `make_stack_layer` only works with dense input. Btw, thanks a lot for this code!

Reply:
thanks for the tip! what is necessary for it to work with sparse input? I'm not familiar with it

Reply:
I just patched the code to `FunctionTransformer(_identity, accept_sparse=True)` to work with sparse input. I'm not sure why you _wouldn't_ want to accept sparse, but there's probably some edge case.